### PR TITLE
allow 5.2 functionality

### PR DESCRIPTION
**Pull request by rwaweber**
https://github.com/logstash-plugins/logstash-codec-nmap/pull/10 seems to have been stationary for a bit now. I would very much like to get functionality for this in 5.2. This is also a want from others as evidenced by https://github.com/logstash-plugins/logstash-codec-nmap/issues/9.

### CODE REVIEW

**Comment by rwaweber** [comment]
@andrewvc Not sure how to pass the CLA check after I've already signed the document, short of whitespace commits.

**Comment by karmi** [comment]
Hi @rwaweber, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/logstash-plugins/logstash-codec-nmap/pull/11.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by karmi** [comment]
Hi @rwaweber, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/logstash-plugins/logstash-codec-nmap/pull/11.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by rwaweber** [comment]
@karmi Apologies, I've added the email now.

File: logstash-codec-nmap.gemspec [hunk 1]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
   s.metadata = { "logstash_plugin" => "true", "logstash_group" => "codec" }
 
   # Gem dependencies
-  s.add_runtime_dependency "logstash-core-plugin-api", "~> 1.0"
+  s.add_runtime_dependency "logstash-core-plugin-api", "~> 1.60", "<= 2.99"
```
Comment (gmoskovicz) [inline logstash-codec-nmap.gemspec:23]: Can you also bump the version in this file to 0.0.17: https://github.com/logstash-plugins/logstash-codec-nmap/blob/master/logstash-codec-nmap.gemspec#L4

Also, can you add the infromation to the changelog?

Thanks!

Comment (gmoskovicz) [reply logstash-codec-nmap.gemspec:23]: Changelog file: https://github.com/logstash-plugins/logstash-codec-nmap/blob/master/CHANGELOG.md

Comment (rwaweber) [reply logstash-codec-nmap.gemspec:23]: Bumped!

**Comment by skoudis** [comment]
When will this be finalized for use with Logstash 5.2?  My project to use this with nmap & OpenVAS is stalled.  Thank you for fixing the plugin.

**Comment by rwaweber** [comment]
@skoudis I think thats up to the maintainer of this plugin. 

In an attempt to alleviate your stalled-ness you could install the plugin manually, but it does take a bit of effort outlined [here](https://github.com/logstash-plugins/logstash-codec-nmap/blob/master/README.md#developing).

I found the simplest means to go about getting jruby ironed out using [rvm](https://rvm.io/rvm/install), then `gem install logstash-devutils bundler` and you should be set to follow the readme from there.

**Comment by suyograo** [comment]
LGTM

**Comment by rwaweber** [comment]
Not sure if I'm doing something incorrectly when trying to install this plugin on my logstash machine, but I'm performing the following with little success, without building the plugin locally.

## Attempt 1
1. `bin/logstash-plugin install logstash-codec-nmap`
2. Yields [this](https://gist.github.com/rwaweber/2c87bfd9c2cd6b238bea64fc38c69646#file-logstash-codec-nmap) (yikes).
3. The important bits [here](https://gist.github.com/rwaweber/2c87bfd9c2cd6b238bea64fc38c69646#file-logstash-codec-nmap-L387) and [here](https://gist.github.com/rwaweber/2c87bfd9c2cd6b238bea64fc38c69646#file-logstash-codec-nmap-L402) seem to indicate that the logstash-plugin repo thinks that the logstash-codec-nmap package is at version 0.

## Attempt 2
1. I added `` to the Gemfile in the root of my logstash directory.
2. `bundle update`
3. Got me a bit [closer](https://gist.github.com/rwaweber/e1c07b2e3f51296221f5586ab71aa13f#file-logstash-codec-nmap2-L20) with some red colorized output, however it seems to think that the latest logstash-codec-nmap package is at version 0.0.15(from the perspective of the logstash-core-plugin-api install process), which is not correct according to the releases [here](https://github.com/logstash-plugins/logstash-codec-nmap/releases). Additionally, the logstash-core install [process](https://gist.github.com/rwaweber/e1c07b2e3f51296221f5586ab71aa13f#file-logstash-codec-nmap2-L11) seems to resolve to 0.0.1 which befuddles me. 

Any clarification or assistance on the above scenarios would be much appreciated! 

## Aside
- Might be necessary to also bump the latest build release to 0.0.17 as well.

**Comment by gmoskovicz** [comment]
@rwaweber seems to be that we didn't publish this yet: https://rubygems.org/gems/logstash-codec-nmap/ . Version 0.0.17 is not there yet, so we need to release it .

**Comment by skoudis** [comment]
@gmoskoviczj  is it possible to publish this soon so we have a fully functional solution to the nmap codec plugin?
Thank you

**Comment by gmoskovicz** [comment]
@suyograo ^^

**Comment by suyograo** [comment]
Published 0.0.17 now

**Comment by TheZeke** [comment]
I've also retried this after the changes and release done apparently 23 hours ago.  I uninstalled and then _re-installed_ logstash completely using apt on Ubuntu 16.04.2 LTS and still get this:

    root@management:/etc/ansible# ssh logstash1-dev
    Welcome to Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-64-generic x86_64)

     * Documentation:  https://help.ubuntu.com
     * Management:     https://landscape.canonical.com
     * Support:        https://ubuntu.com/advantage
    Last login: Wed Mar 22 13:46:32 2017 from 10.10.1.1
    root@logstash1-dev:~# cd /usr/share/logstash/
    root@logstash1-dev:/usr/share/logstash# bin/logstash-plugin install logstash-codec-nmap
    Validating logstash-codec-nmap
    Installing logstash-codec-nmap
    Plugin version conflict, aborting
    ERROR: Installation Aborted, message: Bundler could not find compatible versions for gem "logstash-core-plugin-api":
      In snapshot (Gemfile.lock):
        logstash-core-plugin-api (= 2.1.12)

      In Gemfile:
        logstash-devutils (>= 0) java depends on
          logstash-core-plugin-api (<= 2.99, >= 2.0) java

    [[snip]]

        logstash-input-s3 (>= 0) java depends on
          logstash-mixin-aws (>= 0) java depends on
            logstash-core-plugin-api (<= 2.99, >= 1.60) java

        logstash-codec-nmap (>= 0) java depends on
          logstash-core-plugin-api (~> 1.0) java

        logstash-core-plugin-api (>= 0) java

    Running `bundle update` will rebuild your snapshot from scratch, using only
    the gems in your Gemfile, which may resolve the conflict.
    Bundler could not find compatible versions for gem "logstash-core":
      In snapshot (Gemfile.lock):
        logstash-core (= 5.2.2)

      In Gemfile:
        logstash-core-plugin-api (>= 0) java depends on
          logstash-core (= 5.2.2) java

        logstash-codec-nmap (>= 0) java depends on
          logstash-core (< 3.0.0, >= 2.0.0) java

        logstash-core (>= 0) java

    Running `bundle update` will rebuild your snapshot from scratch, using only
    the gems in your Gemfile, which may resolve the conflict.

**Pull request merged**